src/dos/drive_local.cpp

```diff
@@ -584,13 +588,31 @@ bool localFile::Read(uint8_t *data, uint16_t *size)
 		return false;
 	}
 
+	// Determine if the file is on a disk mounted as floppy or hard drive
+	const int type = local_drive.lock()->GetMediaByte();
+	if (type == 0xF0) {
+		// Floppy drive
+		diskio_delay(*size, floppy_noise.get(), DiskType::Floppy);
```

Comment:
So `size` passed in to this functions gets passed in to the `value` parameter of `diskio_delay`. It's impossible to know what's going on just by looking at these names.

I suggest to do minor refactor here and rename `size` to `num_bytes`, and use `num_bytes` instead of `value` too in `diskio_delay`.